src/test/kotlin/com/ampnet/userservice/controller/RegistrationControllerTest.kt

```diff
@@ -497,6 +565,59 @@ class RegistrationControllerTest : ControllerTestBase() {
         return user
     }
 
+    private fun mockReCaptchaGoogleResponse(status: DefaultResponseCreator, body: String) {
+        val request = ReCaptchaRequest(
+            applicationProperties.reCaptcha.secret,
+            testUser.reCaptchaToken,
+            "127.0.0.1"
+        )
+        mockServer = MockRestServiceServer.createServer(restTemplate)
+        mockServer.expect(
+            ExpectedCount.once(),
+            MockRestRequestMatchers.requestTo(applicationProperties.reCaptcha.url)
+        )
+            .andExpect(MockRestRequestMatchers.method(HttpMethod.POST))
+            .andExpect(
+                MockRestRequestMatchers.content()
+                    .contentType(MediaType.APPLICATION_JSON)
+            )
+            .andExpect(MockRestRequestMatchers.content().json(objectMapper.writeValueAsString(request)))
+            .andRespond(status.body(body))
+    }
+
+    private fun generateSuccessfulGoogleResponse(): String =
+        """
+            {
+                "success":"true",
+                "challenge_ts":"56743453",
+                "hostname":"user_hostname",
+                "score": "0.6",
+                "error-codes": []
+            }
+        """.trimIndent()
+
+    private fun generateUnSuccessfulGoogleResponse(): String =
+        """
+            {
+                "success":"false",
+                "error-codes": [
+                    "missing-input-response",
+                    "missing-input-secret"
+                ]
+            }
+        """.trimIndent()
+
+    private fun generateLowScoreGoogleResponse(): String =
+        """
+            {
+                "success":"true",
+                "challenge_ts":"56743453",
+                "hostname":"user_hostname",
+                "score": "0.4",
+                "error-codes": []
+            }
+        """.trimIndent()
+
```

Comment:
In RegistrationControllerTest use mocked ReCaptcha service. Create ReCaptchaServiceTest and move this verification.